File: app/src/main/java/io/github/howiezuo/unsplash/binding/PhotosBinding.java
package io.github.howiezuo.unsplash.binding;

import android.databinding.BindingAdapter;
import android.support.v7.widget.RecyclerView;

import java.util.List;

import io.github.howiezuo.unsplash.feature.MainAdapter;
import io.github.howiezuo.unsplash.model.Photo;

public class PhotosBinding {

    @BindingAdapter("bind:items")
    public static void setItems(RecyclerView recyclerView, List<Photo> photos) {
        MainAdapter adapter = (MainAdapter) recyclerView.getAdapter();
        if (adapter != null) {
            adapter.setDataSet(photos);
        }
    }

}

File: app/src/main/java/io/github/howiezuo/unsplash/binding/PhotosBinding.kt
package io.github.howiezuo.unsplash.binding

import android.databinding.BindingAdapter
import android.support.v7.widget.RecyclerView
import io.github.howiezuo.unsplash.feature.main.MainAdapter
import io.github.howiezuo.unsplash.model.Photo


object PhotosBinding {

    @JvmStatic
    @BindingAdapter("bind:items")
    fun setItems(recyclerView: RecyclerView, photos: List<Photo>) {
        val adapter: MainAdapter? = recyclerView.adapter as? MainAdapter
        adapter?.setDateSet(photos)
    }

}
